core/controllers/ingest.py

```diff
@@ -10,28 +10,49 @@
 from core.const import EXCEL_MIMETYPE
 from core.controllers.mappings import INGEST_MAPPINGS, DataMapping
 from core.db import db
+from core.db.entities import Submission
 from core.errors import ValidationError
+from core.extraction.towns_fund import ingest_towns_fund_data
 from core.validation.casting import cast_to_schema
+from core.validation.failures import InvalidEnumValueFailure
 from core.validation.validate import validate
 
+ETL_PIPELINES = {
+    "tf_round_three": ingest_towns_fund_data,
+}
+
 
 def ingest():
-    """Ingests spreadsheet data and stores it in a database.
+    """Ingests a spreadsheet submission and stores its contents in a database.
 
     This function takes in an Excel file object and extracts data from the file, casts it to the
-    specified schema, and validates it. If the data fails validation, a `ValidationError` is raised. Otherwise, the
-    data is cleaned and ingested into a database.
+    specified schema, and validates it.
+    If source_type is specified, the data will be put through an etl pipeline, prior
+    to validation.
+    If the data fails validation, a `ValidationError` is raised. Otherwise, the data is cleaned and ingested into a
+    database.
 
     :return: A tuple containing a string message and an integer HTTP status code.
     :raises ValidationError: If the data fails validation against the specified schema.
     """
     excel_file = connexion.request.files["excel_file"]  # required
+    source_type = connexion.request.form.get("source_type")  # required
```

Comment:
Are these not passed through as method args rather than having to fetch them from connexion? e.g. as per: https://connexion.readthedocs.io/en/stable/request.html#automatic-parameter-handling ?